### PR TITLE
BigQuery: Adds Python 3.7 and removes Python 3.4

### PR DESCRIPTION
**Pull request by alixhami**
Resolves https://github.com/GoogleCloudPlatform/google-cloud-python/issues/5294

### CODE REVIEW

File: bigquery/nox.py [hunk 1]
```diff
@@ -65,7 +65,7 @@ def default(session):
 
 
 @nox.session
-@nox.parametrize('py', ['2.7', '3.5', '3.6'])
+@nox.parametrize('py', ['2.7', '3.5', '3.6', '3.7'])
```
Comment (tswast) [inline bigquery/nox.py]: Add the same to `system`?

Comment (tswast) [reply bigquery/nox.py]: I guess we don't have 3.5 on system, either, nevermind.

Comment (tseaver) [reply bigquery/nox.py]: Right, we need to keep `system` as snappy as possible.  The chance that 3.x will break a system test which 3.y passes is pretty minimal.

Comment (crwilcox) [reply bigquery/nox.py]: System should likely be set to 2.7, 3.7 though.

**Comment by tswast** [comment]
Right, wasn't that what https://github.com/GoogleCloudPlatform/google-cloud-python/pull/5295 was about? Or did I misinterpret the intent?

File: bigquery/nox.py (hunk 1 above)
Comment (tseaver) [reply bigquery/nox.py]: 3.7 is still beta.

**Pull request merged**